contrib/database/sql/option.go

```diff
@@ -74,3 +76,13 @@ func WithDSN(name string) Option {
 		cfg.dsn = name
 	}
 }
+
+// WithCheckSpanContext enables checking the provided context
+// for a parent span before creating one for the request.
+// With this enabled, no span is created for the request
+// without an existing parent span.
+func WithCheckSpanContext() Option {
```

Comment:
How about something like `WithChildSpanOnly`? `WithCheckSpanContext` says too much about the implementation and not enough about what the effect on the integration is.

We can also simplify the documentation. Since it's user-facing, we don't need to detail the implementation that much. How about just
> [name] causes spans to be created only when there is an
existing parent span in the Context.